ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java

```diff
@@ -405,6 +405,17 @@ static boolean installSnapshotEnabled(RaftProperties properties) {
       static void setInstallSnapshotEnabled(RaftProperties properties, boolean shouldInstallSnapshot) {
         setBoolean(properties::setBoolean, INSTALL_SNAPSHOT_ENABLED_KEY, shouldInstallSnapshot);
       }
+
+      String READ_STATEMACHINE_DATA_TIMEOUT_KEY = PREFIX + ".read.statemachine.data.timeout";
+      TimeDuration READ_STATEMACHINE_DATA_TIMEOUT_DEFAULT = TimeDuration.valueOf(1000, TimeUnit.MILLISECONDS);
+      static TimeDuration readStatemachineDataTimeout(RaftProperties properties) {
+        return getTimeDuration(properties.getTimeDuration(READ_STATEMACHINE_DATA_TIMEOUT_DEFAULT.getUnit()),
+            READ_STATEMACHINE_DATA_TIMEOUT_KEY, READ_STATEMACHINE_DATA_TIMEOUT_DEFAULT, getDefaultLog());
+      }
+      static void setReadStatemachineDataTimeout(RaftProperties properties, TimeDuration readStatemachineTimeout) {
+        setTimeDuration(properties::setTimeDuration, READ_STATEMACHINE_DATA_TIMEOUT_KEY, readStatemachineTimeout);
+      }
+
```

Comment:
Let's move the new conf next to sync.timeout in RaftServerConfigKeys.Log.StateMachineData.
```
      String READ_TIMEOUT_KEY = PREFIX + ".read.timeout";
      TimeDuration READ_TIMEOUT_DEFAULT = TimeDuration.valueOf(1000, TimeUnit.MILLISECONDS);
      static TimeDuration readTimeout(RaftProperties properties) {
        return getTimeDuration(properties.getTimeDuration(READ_TIMEOUT_DEFAULT.getUnit()),
            READ_TIMEOUT_KEY, READ_TIMEOUT_DEFAULT, getDefaultLog());
      }
      static void setReadTimeout(RaftProperties properties, TimeDuration readStatemachineTimeout) {
        setTimeDuration(properties::setTimeDuration, READ_TIMEOUT_KEY, readStatemachineTimeout);
      }
```